app.ts

```diff
@@ -158,8 +190,22 @@ const releaseBuildNumber = (() => {
     return '';
 })();
 
+export type AppDefaultArguments = {
+    rootDir: string;
+    env: string[];
+    hostname?: string;
+    port: number;
+    gitReleaseName: string;
+    releaseBuildNumber: string;
+    wantedLanguages: string | null;
+    doCache: boolean;
+    fetchCompilersFromRemote: boolean;
+    ensureNoCompilerClash: boolean | undefined;
+    suppressConsoleLog: boolean;
+};
```

Comment:
Perhaps it would be better to refactor the above Partial<{...}> type into a specific one, and instead type defArgs as Partial of that one too

Reply:
At the moment there is not sufficient overlap between the opts type and the defArgs type and unlike the opts from nopt, for defArgs we do actually know exactly what members are present so it's nice to have it non-partial.

Reply:
I'm going to go ahead and merge, can refactor in a later PR if it'd be good to do so